internal/common/lxd_exec.go

```diff
@@ -39,9 +40,8 @@ func (e *ExecModel) Execute(ctx context.Context, server lxd.InstanceServer, inst
 		return diags
 	}
 
-	// If command is one liner, split it on spaces.
-	if len(command) == 1 {
-		command = strings.SplitN(command[0], " ", 3)
+	if e.Interpreter.ValueString() == "shell" {
+		command = append([]string{"/bin/sh", "-c"}, command...)
```

Comment:
I don't like hardcoding the interpreter to `/bin/sh`. It assumes that this path will exist, which is not necessarily true, and can be a different package depending on the image being worked with. Could we instead allow a user to provide their own interpreter rather than selecting a generic `shell` value?

Reply:
Yeah, by default we do not use any interpreter, so the user can provide the command as:
```hcl
command = ["/bin/sh", "-c", "..."]
```

I though about having an interpreter as a list of strings (terraform's [local-exec](https://developer.hashicorp.com/terraform/language/resources/provisioners/local-exec#interpreter) uses that approach):
```hcl
command     = ["..."]
interpreter = ["/bin/sh", "-c"]
```

But it seems like just splitting a list into two lists, or? 

Could we maybe have `interpreter` attribute as you have proposed, but have `command` as a string? - However, how would then a simple command like `apt update` look like if we want to avoid using shell?
```hcl
command     = "apt update"
interpreter = ["sh", "-c"]

command     = "update"
interpreter = ["apt"]
```

Maybe the interpreter *should* be the required field?

Reply:
If you take that route, might as well use this form:
```
command   = "apt" # note the bare string, not an array/list
arguments = ["dist-upgrade", "-V", "-y"]
```

Reply:
Some prior art would be the built-in local-exec provisioner. It sets a default of `/bin/sh -c ` (in a slice) if there is not a specified parameter and takes a simple string as the command. `working_dir` and `environment` may not be bad to offer either.

https://developer.hashicorp.com/terraform/language/resources/provisioners/local-exec
https://github.com/hashicorp/terraform/blob/main/internal/builtin/provisioners/local-exec/resource_provisioner.go

I'd be fine with defaulting to `/bin/sh` if there is an alternative

Reply:
How about we have three attributes? - `posix`, `command`, `interpreter`

posix would be mutually exclusive with command and interpreter.
*I don't like posix as attribute name, but just for demo*

So user can provide either:
`posix = ["ls", "/"]` // So that we allow executing commands without shell

or:
```hcl
command     = "ls /"
interpreter = ["bash", "-c"] # Defaults to /bin/sh -c
```

With this approach, we could maybe also have list of `commands` instead of single command per exec block?

Reply:
@terraform-lxd/maintainers Any comments/suggestions on https://github.com/terraform-lxd/terraform-provider-lxd/pull/386#discussion_r1421009176?

Reply:
I think my preference would be just using `command` and `interpreter`, with `interpreter` defaulting to `bash -c` and `command` being a string. This should support simple cases of just wanting to run a command and not worrying about the interpreter, but also allow for more complex cases where specifying the interpreter is desirable. I'm not sure if I see the value in also having a `posix` attribute, unless I'm missing something.

Reply:
Thanks everyone. 

Based on the comments, I will leave `command` as is (list of strings), and change the `interpreter` attribute from string (enum) to list of strings. If interpreter is set, it will be simply prepended to the command.